### PR TITLE
[OVC] do not parse inputs for py_api

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
In cases like below ovc should not parse input and an error should be thrown:
 ```
from openvino.tool.ovc import ocnvert_model
...
 ov_model = ov.convert_model('model.onnx', input="input.1[5,3,224,224]")
```
 
### Tickets:
 - [CVS-119474](https://jira.devtools.intel.com/browse/CVS-119474)

### CODE REVIEW

File: tools/ovc/openvino/tools/ovc/cli_parser.py [hunk 1]
```diff
@@ -116,7 +110,20 @@ def is_single_input(input: [tuple, list]):
...
+    for input_value in split_inputs(inputs):
+
+        # Parse string with parameters for single input
+        node_name, shape = parse_input_value(input_value)
+        # pylint: disable=no-member
+        inputs_list.append((node_name, shape))
+    return inputs_list
+
+
+def input_to_input_cut_info(input: [dict, tuple, list]):
```
Comment (popovaan) [inline tools/ovc/openvino/tools/ovc/cli_parser.py:125]: This method still parses input_value in 140 line. 
Looks like we need to remove lines 134-144.

File: tests/layer_tests/ovc_python_api_tests/test_pytorch.py [hunk 2]
```diff
@@ -1093,6 +1093,21 @@ def test_example_inputs(self):
...
+        with self.assertRaisesRegex(Exception, ".*No node with name.*"):
+            convert_model(pytorch_model, input='input1[1, 10]')
+
+    def test_incorrect_inputs_2(self):
+        from openvino.tools.ovc import convert_model
+        pytorch_model, _, _ = create_pytorch_nn_module_case1('')
+
+        # check that it accepts specified names as is without parsing into 2 different inputs
+        with self.assertRaisesRegex(Exception, 'No node with name input1\[1, 10\],input2\[2, 100\]'):
+            convert_model(pytorch_model, input='input1[1, 10],input2[2, 100]')
```
Comment (popovaan) [inline tests/layer_tests/ovc_python_api_tests/test_pytorch.py:1109]: Please add negative test for
convert_model(pytorch_model, input='input1,input2')

File: tools/ovc/openvino/tools/ovc/cli_parser.py (hunk 1 above)
Comment (pavel-esir) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:125]: oouups, i actually removed them, but here they left by mistake. Will remove them

Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:125]: Corrected my comment. Let's not remove them, but replace with the suggested code. As single string should be treated as single node name.

File: tests/layer_tests/ovc_python_api_tests/test_pytorch.py (hunk 2 above)
Comment (popovaan) [reply tests/layer_tests/ovc_python_api_tests/test_pytorch.py:1117]: And this negative test:
convert_model(pytorch_model, input='input1[1,10]','input2[2,100]')

File: tools/ovc/openvino/tools/ovc/cli_parser.py (hunk 1 above)
Comment (pavel-esir) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:125]: removed this lines.
discussed offline, after all i decided to leave checking if it's a string outside of `input_to_input_cut_info`, e.g.
https://github.com/openvinotoolkit/openvino/blob/372009b6f02aa6300c616895bd45ce107c46f849/tools/ovc/openvino/tools/ovc/convert_impl.py#L97-L101

File: tests/layer_tests/ovc_python_api_tests/test_pytorch.py (hunk 2 above)
Comment (pavel-esir) [reply tests/layer_tests/ovc_python_api_tests/test_pytorch.py:1117]: done

Comment (pavel-esir) [reply tests/layer_tests/ovc_python_api_tests/test_pytorch.py:1117]: added test for `convert_model(pytorch_model, input=['input1[1,10]','input2[2,100]'])` as well.

**Review by popovaan — APPROVED** [review]

File: src/bindings/python/src/openvino/frontend/tensorflow/utils.py [hunk 3]
```diff
@@ -264,10 +266,19 @@ def tf_function(args):
 
         input_needs_packing = True
 
+    def are_shapes_defined(shape: Union[List, Dict]):
+        if shape is None:
+            return False
+
+        if isinstance(shape, list):
```
Comment (rkazants) [inline src/bindings/python/src/openvino/frontend/tensorflow/utils.py:273]: What will be if the list is empty?

Comment (pavel-esir) [reply src/bindings/python/src/openvino/frontend/tensorflow/utils.py:273]: Agreed, better to add a check. Normally we should not have an empty list argument here: if user specified input=[] then `normalize_inputs` will convert them to `None`
https://github.com/openvinotoolkit/openvino/blob/372009b6f02aa6300c616895bd45ce107c46f849/tools/ovc/openvino/tools/ovc/convert_impl.py#L356

But if something goes wrong and this function is called with empty list from somewhere else then there will be a problem because
```
In [4]: np.all([])
Out[4]: True
```
it will try to create input tensors but it should not do that for empty shapes. Therefore added check for emptiness as well.

File: tools/ovc/openvino/tools/ovc/cli_parser.py [hunk 4]
```diff
@@ -37,18 +37,15 @@ def is_shape_type(value):
...
     :param input: input cut parameters of single input
     :return: InputCutInfo
     """
     if isinstance(input, str):
-        # Parse params from string
-        node_name, shape = parse_input_value(input)
         # pylint: disable=no-member
-        return _InputCutInfo(node_name,
-                             PartialShape(shape) if shape is not None else None)
+        return _InputCutInfo(input, None)
```
[the remark this answers is not in the document]
Comment (pavel-esir) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:48]: Returned back `str` to signature. I removed them in the first place because implemented solution without strings, but it turned out string args are still necessary but i forgot to return them back to signature. Thanks for noticing
